lib/listPackages.js

```diff
@@ -2,55 +2,52 @@ import { exec } from "@actions/exec";
 import npmArgs from "./npmArgs.js";
 
 /**
- * @param {import("@actions/exec").ExecOptions?} options
+ * @param {import("@actions/exec").ExecOptions} [options]
  * @returns {Promise<Map<string, string>>}
  */
 export default async function listPackages(options = {}) {
+  const cwd = options.cwd || process.cwd();
   let lines = "";
   let stderr = "";
-  const returnCode = await exec("npm", npmArgs("ls", "--parseable", "--long", "--all"), {
-    listeners: {
-      stdout: (data) => {
-        lines += data.toString();
+  const returnCode = await exec(
+    "npm",
+    npmArgs("ls", "--parseable", "--long", "--all", "--package-lock-only"),
```

Comment:
[note] `--package-lock-only` does not scan `node_modules`. See:
https://docs.npmjs.com/cli/v7/commands/npm-ls#package-lock-only